rpcserver.go

```diff
@@ -3013,13 +3013,30 @@ func SignRawTransaction(w *wallet.Wallet, chainSvr *chain.Client, icmd interface
 			err = vm.Execute()
 		}
 		if err != nil {
+			localErr := err
+			class, addr, _, _ := txscript.ExtractPkScriptAddrs(
+				inputs[txIn.PreviousOutPoint],
+				activeNet.Params)
+
+			if err == txscript.ErrStackUnderflow &&
+				class == txscript.ScriptHashTy {
+				redeemScript, _ := getScript(addr[0])
+				redeemClass, _, _, _ := txscript.ExtractPkScriptAddrs(
```

Comment:
It's much cheaper to use `txscript.GetScriptClass` since all you're after is the class of the script.